src/auth0/databaseConnections.js

```diff
@@ -0,0 +1,128 @@
+    return database.name;
+  });
+
+  return apiCall(client, client.connections.getAll, [ { strategy: 'auth0' } ])
+    .then(function(connections) {
+      progress.connections = connections.filter(function(connection) {
+        return databaseNames.indexOf(connection.name) > -1;
+      });
+      return progress.connections;
+    });
+};
+
+/*
+ * Update a database.
+ */
+const updateDatabase = function(progress, client, connections, database) {
+  var allowedScripts = null;
+  progress.log('Processing connection ' + database.name);
+
+  const connection = _.find(connections, { name: database.name });
+  if (!connection) {
+    return Promise.reject(
+      new ValidationError('Unable to find connection named: ' + database.name)
+    );
+  }
+
+  const options = connection.options || {};
+  options.customScripts = {};
+
+  const databaseScriptKeys = Object.keys(database.scripts);
+
+  allowedScripts = (options.import_mode) ? constants.DATABASE_SCRIPTS_IMPORT : constants.DATABASE_SCRIPTS_NO_IMPORT;
+  /* Check if change_email is included and if it is, allow get_users for non-import */
+  if (!options.import_mode) {
+    if (databaseScriptKeys.indexOf(constants.DATABASE_SCRIPTS_CHANGE_EMAIL) >= 0) {
+      if (databaseScriptKeys.indexOf(constants.DATABASE_SCRIPTS_GET_USER) >= 0) {
+        allowedScripts = constants.DATABASE_SCRIPTS;
+      } else {
+        throw new ValidationError('The ' + constants.DATABASE_SCRIPTS_CHANGE_EMAIL + ' script requires the ' + constants.DATABASE_SCRIPTS_GET_USER + ' script for ' + database.name + '.');
+      }
+    }
+  }
+  progress.log('Import User to Auth0 enabled: ' + options.import_mode + '. Allowed scripts: ' + JSON.stringify(allowedScripts, null, 2));
+
+  // Set all custom scripts
+  _(databaseScriptKeys).forEach(function(scriptName) {
+    if (allowedScripts.indexOf(scriptName) < 0) {
+      throw new ValidationError('The ' + scriptName + ' script is not allowed for ' + database.name + '.');
+    }
+
+    if (!database.scripts[scriptName].scriptFile || database.scripts[scriptName].scriptFile.length === 0) {
+      throw new ValidationError('The ' + scriptName + ' script for ' + database.name + ' is empty.');
+    }
+
+    options.customScripts[scriptName] = database.scripts[scriptName].scriptFile;
+  });
+
+  progress.connectionsUpdated += 1;
+  progress.log('Updating database ' + connection.id + ': ' + JSON.stringify(options, utils.checksumReplacer(Object.keys(options.customScripts)), 2));
+  return apiCall(client, client.connections.update, [ { id: connection.id }, { options: options } ]);
+};
+
+/*
+ * Update all databases.
+ */
+const updateDatabases = function(progress, client, databases) {
+  if (databases.length === 0) {
+    return Promise.resolve(true);
+  }
+
+  return getDatabaseConnections(progress, client, databases)
+    .then(function(connections) {
+      return Promise.map(databases,
+        function(database) { return updateDatabase(progress, client, connections, database); },
```

Comment:
I think we need to add a meta data file for enabling connections and *not* allow enabling connections directly in the JSON.  If we want this to work seamlessly we need to be able to specify a client *name* instead of *id* when enabling connections, otherwise we don't have a good way to do it without having to run it once, create the clients, then get all the client ID's, update the config.json and run again.

Reply:
Overall this is a good idea, but this is a much broader request. Using a metadata file to map client names to id's, won't work if clients share the same name (which is possible). So, this needs some analysis. @mostekcm WDYT?

Reply:
We already *enforce* one client per name in the client's section.  Even though Auth0 supports it, we don't in the deploy CLI